src/XGridCtrl.cpp

```diff
@@ -564,6 +564,11 @@ XGridCtrl::DrawSquare(wxDC & dc, const puz::Square & square, const wxColour & co
         m_drawer.RemoveFlag(XGridDrawer::DRAW_OUTLINE);
         m_drawer.AddFlag(XGridDrawer::DRAW_FLAG | XGridDrawer::DRAW_NUMBER);
     }
+
+    if (propagate && square.GetPartnerSquare()) {
+        puz::Square* partner = square.GetPartnerSquare();
+        DrawSquare(dc, *partner, GetSquareColor(*partner), false);
```

Comment:
Looks like DrawSquare is only ever called with either wxNullColor or with the result of GetSquareColor (or a precalculated color based on what GetSquareColor should return). I'd guess if we're called with wxNullColor we probably want to pass wxNullColor on to the partner square, but otherwise use GetSquareColor like you have here.

Reply:
When I make this change, upon first opening a puzzle, only the single selected square in the quote (the initial selected word) ends up being highlighted. The entire quote "word" is supposed to be colored appropriately as a selected word. If I switch words to/from something else, it works fine, but this doesn't seem to be a safe change to make as is.

Reply:
Huh, I see that. I think the issue is that `wxNullColour` is being used to mean two different things:

1. I don't know what the color should be (in GetSquareColor)
2. I know this used to be highlighted, but now erase the highlight (in SetFocusedSquare)

And I think this only ends up being a problem when we're doing incremental drawing with wxClientDC instead of letting OnPaint handle the drawing, but here we are.

I messed around with this a little and came up with add235c. See if that works for you.

Reply:
Cherry-picked that commit into my branch and it seems to work fine here! I can append to this PR, or I guess you can just merge that branch, if you're happy.

Reply:
Great! Either ways is fine by me. I won't get to it today, but by tomorrow I'm going to try to prepare a release, so if it's in this PR I'll merge, otherwise I can cherry-pick to master afterwards.

Reply:
Added your commit to the merge. Excited to hear a release is coming soon!